### DISCUSSION TITLE
Return value from a scope

### DISCUSSION THREAD

**Discussion by rccarlson** [dotnet/csharplang#7836] [Language Ideas]
I've been learning F# and I've really enjoyed the scoped definitions available there. For example:
```fsharp
let sum =
  let a = 1
  let b = 2
  a + b
```
This would set `sum` to equal `3`, while leaving `a` and `b` out of scope for any other code. I would love to have something like this for C#. My vision for this would be something like:
```csharp
int sum = {
	int a = 1;
	int b = 2;
	return a + b;
};
```
This would have the same behavior as:
```csharp
int sum = new Func<int>(()=>{
	int a = 1;
	int b = 2;
	return a + b;
}).Invoke();
```

This would be very helpful for cleaning up working variables (such as `StringBuilder`s or iterators) and only holding on to the resulting value

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/issues/3086
